apiserver/httpcontext.go

```diff
@@ -111,9 +111,65 @@ func checkPermissions(tag names.Tag, acceptFunc common.GetAuthFunc) (bool, error
 	return false, errors.NotValidf("tag kind %v", tag.Kind())
 }
 
-// stateForRequestAuthenticatedUser is like stateForRequestAuthenticated
+// stateForMigration asserts that the incoming connection is from a user that
+// has admin permissions on the controller model. The method also gets the
+// model uuid for the model being migrated from a request header, and returns
+// the state instance for that model.
+func (ctxt *httpContext) stateForMigration(r *http.Request) (st *state.State, err error) {
+	var user state.Entity
+	st, user, err = ctxt.stateAndEntityForRequestAuthenticatedUser(r)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			ctxt.release(st)
+		}
+	}()
+
+	if !st.IsController() {
+		return nil, errors.BadRequestf("model is not controller model")
+	}
+	admin, err := st.IsControllerAdmin(user.Tag().(names.UserTag))
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !admin {
+		return nil, errors.Unauthorizedf("not a controller admin")
+	}
+
+	modelUUID, err := validateModelUUID(validateArgs{
+		statePool: ctxt.srv.statePool,
+		modelUUID: r.Header.Get("Migration-Model-UUID"),
```

Comment:
I think this should be X-Juju-Migration-Model-UUID for consistency with X-Juju-Nonce from `apiserver/params/constants.go`.